firefox-ios/Ecosia/Core/HTTPClient/URLSessionProtocol.swift

```diff
@@ -6,6 +6,7 @@ import Foundation
 
 public protocol URLSessionProtocol {
     func data(from url: URL) async throws -> (Data, URLResponse)
+    func data(for request: URLRequest, delegate: (any URLSessionTaskDelegate)?) async throws -> (Data, URLResponse)
```

Comment:
Why did this have to be added?

Reply:
In order to promote the possibility to pass the `URLRequest` directly instead of the `URL`.  But then I updated the implementation to deal with the `HTTPClient` directly so no need to keep this one any longer!